### ISSUE TITLE
Show workspaceClasses display names and extend config with credits

### ISSUE THREAD

**Issue by jankeromnes** [gitpod-io/gitpod#13362]
## Is your feature request related to a problem? Please describe

Show workspace classes' display names instead of their internal names in the Usage view.

## Describe the behaviour you'd like

E.g., instead of showing `g1-large` (the internal name), we should show `Large` (the display name).

Internal names and display names are configured [here](https://github.com/gitpod-io/ops/blob/208e9f7b2dc264c68a01a89ef763de43f680315e/deploy/production/meta-eu02/app/installer-config.yaml#L186-L218) (internal)

Tasks
- [x] https://github.com/gitpod-io/ops/pull/6434
- [x] https://github.com/gitpod-io/ops/pull/6597
- [x] https://github.com/gitpod-io/gitpod/pull/14327
- [x] https://github.com/gitpod-io/gitpod/pull/14362
- [x] https://github.com/gitpod-io/ops/pull/6877

**Comment by laushinka** [comment]
What do "old" mean from the users' perspective?

**Comment by easyCZ** [comment]
We currently have two datasets that define information relating to Workspace Classes.
1. [config](https://github.com/gitpod-io/ops/blob/208e9f7b2dc264c68a01a89ef763de43f680315e/deploy/production/meta-eu02/app/installer-config.yaml#L186-L218)
2. [usage config](https://github.com/gitpod-io/ops/blob/208e9f7b2dc264c68a01a89ef763de43f680315e/deploy/production/meta-eu02/app/installer-config.yaml#L177-L183) which defines a subset of the config in 1.

There's a high chance that sooner or later this config will drift. Ideally, we'd derive this from the same source of truth.

The problem is that there are varying guarantees:
1. The `workspaceClasses` config (owned by Team Workspace) does not provide guarantees on keeping old definitions of classes around. Old classes may be removes.
2. Usage config maps the WorkspaceClass ID to a price. It requires that old Workspace Class to continue being defined because the Usage View shows historic data - in a way, we can never remove a workspace class.

**A possible approach**
We include the credits price on the `workspaceClasses` definition. This means we can derive our information for the usage component from a single source of truth. For older workspace classes which are in the DB but not in the definition we'd continue to show the WorkspaceClass.ID as a fall-back.

Are we happy with extending the workspacClasses definition with credits? In a way, it makes sense as defining a workspaceClass without credit cost is basically incompatible with the existing system already.

**Comment by easyCZ** [comment]
@gitpod-io/engineering-workspace What's your appetite for including `credits` in the definition of a Workspace Class?

**Comment by Furisto** [comment]
@easyCZ Feel free to add it

**Comment by kylos101** [comment]
@Furisto I assume you mean feel free to add like @easyCZ shared?
 1) [add it to the installer config](https://github.com/gitpod-io/ops/blob/208e9f7b2dc264c68a01a89ef763de43f680315e/deploy/production/meta-eu02/app/installer-config.yaml#L186-L218) 
2) parse the installer config to [add to the usage config](https://github.com/gitpod-io/ops/blob/208e9f7b2dc264c68a01a89ef763de43f680315e/deploy/production/meta-eu02/app/installer-config.yaml#L177-L183)

In other words, we would not parse the installer config and also add it to `ws-manager` or `workspace-templates` config, because we do not have a need to use workspace class usage info with those components.

**Comment by jankeromnes** [comment]
(Note in passing: After reading all the comments, it seems that the scope of this issue has increased from:

> Show correct labels in UI

to:

> Show correct labels in UI + Improve workspace classes configuration by merging class and price specifications

Both of these could be addressed by the same PR, but maybe we should update the issue title to reflect the new scope 💭)

**Comment by laushinka** [comment]
> maybe we should update the issue title to reflect the new scope?

💯 Will update the title to also include extending the workspaceClasses definition with the credits.

**Comment by laushinka** [comment]
@Furisto @jldec What credits info should we include in the workspaceClasses config? ([internal thread](https://gitpod.slack.com/archives/C02EN94AEPL/p1667290331418909))

**Comment by jldec** [comment]
> What credits info should we include in the workspaceClasses config

@easyCZ - since you added the [comment](#issuecomment-1275724308) could you clarify what needs to be added so that we can draw the necessary details from a single config (and avoid configuration drift)

**Comment by laushinka** [comment]
After the discussion in [this thread](https://gitpod.slack.com/archives/C02EN94AEPL/p1667290331418909), I will add the credits info [this way](https://github.com/gitpod-io/ops/pull/6434/commits/e454a2842c078ca3d50304da5da1e7c9e34f33f0) under `workspaceClasses` - with the field `perMinute`. This way if we ever need to add other units (e.g. `hourly`), we can do so.

**Comment by laushinka** [comment]
> @laushinka What's missing to move forward with this issue? Asking because the status is not clear to me. 🤔

Thanks for the nudge! Sorry, I was sure I had moved the card to `In Validation`. Will do that! Status is making sure this week that the new config is working well and we can remove the old one 🙏🏽
